### PR TITLE
4670339: (ch) Channels.newChannel(OutputStream) should flush stream on every write

### PR DESCRIPTION
**Pull request by bplb**
Call `flush()` on the `OutputStream` before returning from `WritableByteChannelImpl::write`.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change requires a CSR request matching fixVersion 21 to be approved (needs to be created)

### Issue
 * [JDK-4670339](https://bugs.openjdk.org/browse/JDK-4670339): (ch) Channels.newChannel(OutputStream) should flush stream on every write

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13198/head:pull/13198` \
`$ git checkout pull/13198`

Update a local copy of the PR: \
`$ git checkout pull/13198` \
`$ git pull https://git.openjdk.org/jdk.git pull/13198/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13198`

View PR using the GUI difftool: \
`$ git pr show -t 13198`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13198.diff">https://git.openjdk.org/jdk/pull/13198.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13198#issuecomment-1486005048)

### CODE REVIEW

**Comment by AlanBateman** [comment]
This changes long standing behavior and may impact performance when the output stream is buffered. Is there a compelling reason to change the behavior? If so then maybe other options could be explored, `newChannel(OutputStream, boolean autoFlush)` or just documenting that automatic flushing can be done by wrapping the output stream.

File: test/jdk/java/nio/channels/Channels/Basic.java [hunk 1]
```diff
@@ -288,6 +298,32 @@ private static void testNewChannelOut(File blah) throws Exception {
         fos.close();
     }
 
+    private static void testNewChannelOutFlush() throws Exception {
+        final int size = 8192;
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream(size);
+        try (BufferedOutputStream bos = new BufferedOutputStream(baos, size);
+             WritableByteChannel wbc = Channels.newChannel(bos, true);) {
```
Comment (turbanoff) [inline test/jdk/java/nio/channels/Channels/Basic.java:306]:
```suggestion
             WritableByteChannel wbc = Channels.newChannel(bos, true)) {
```

File: src/java.base/share/classes/java/nio/channels/Channels.java [hunk 2]
```diff
@@ -342,7 +343,8 @@ protected void implCloseChannel() throws IOException {
      *
      * <p> The resulting channel will not be buffered; it will simply redirect
      * its I/O operations to the given stream.  Closing the channel will in
-     * turn cause the stream to be closed.  </p>
+     * turn cause the stream to be closed.  The returned channel implements
+     * {@linkplain Flushable}; flushing the channel will flush the stream. </p>
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/nio/channels/Channels.java:347]: This was just a suggestion to try but I think it's a bit too weird so I would drop that. 

I think the main thing we need to decide is whether a factory method for an auto-flushing output stream is useful to have to not. Maybe we should try to survey the usages of this factory method to see if there is much usage and in what contexts.

**Comment by bplb** [comment]
Withdrawing this pull request as the requested functionality is easily enough to implement outside the JDK as illustrated by the example code in the associated issue [JDK-4670339](https://bugs.openjdk.org/browse/JDK-4670339)

**Pull request closed without merging**